common/src/main/java/io/netty/util/AsciiString.java

```diff
@@ -714,6 +714,34 @@ public int indexOf(CharSequence subString, int start) {
     }
 
     /**
+     * Searches in this string for the index of the specified char {@code ch}.
+     * The search for the char starts at the specified offset {@code start} and moves towards the end of this string.
+     *
+     * @param ch the char to find.
+     * @param start the starting offset.
+     * @return the index of the first occurrence of the specified char {@code ch} in this string,
+     * -1 if found no occurrence.
+     */
+    public int indexOf(char ch, int start) {
+        if (start < 0) {
```

Comment:
@Spikhalskiy - being consistent sgmt.

However I'm not sure that I'm clear on terminology. By "safe" I assume you mean does not throw, and will modify input if necessary to run "as expected" (less than start gets set to start, greater than length return `-1`)? It looks like the `String.indexOf` member function is "safe" by this definition:

```java
public int indexOf(int ch, int fromIndex) {
        final int max = value.length;
        if (fromIndex < 0) {
            fromIndex = 0;
        } else if (fromIndex >= max) {
            // Note: fromIndex might be near -1>>>1.
            return -1;
        }
...
}
```